server/plugin/api.go

```diff
@@ -1188,55 +1184,74 @@ func (p *Plugin) handlePipelineApproveOrRejectRequest(w http.ResponseWriter, r *
 	organization := postActionIntegrationRequest.Context[constants.PipelineRequestContextOrganization].(string)
 	projectName := postActionIntegrationRequest.Context[constants.PipelineRequestContextProjectName].(string)
 	approvalID := int(postActionIntegrationRequest.Context[constants.PipelineRequestContextApprovalID].(float64))
-	statusCode, err := p.Client.UpdatePipelineApprovalRequest(pipelineApproveRequestPayload, organization, projectName, mattermostUserID, approvalID)
+	statusCode, updatePipelineApprovalRequestErr := p.Client.UpdatePipelineApprovalRequest(pipelineApproveRequestPayload, organization, projectName, mattermostUserID, approvalID)
 	switch statusCode {
 	case http.StatusOK:
-		post, _ := p.API.GetPost(postActionIntegrationRequest.PostId)
-		slackAttachment := post.Attachments()[0]
-		slackAttachment.Actions = nil
-		slackAttachment.Fields = []*model.SlackAttachmentField{
-			slackAttachment.Fields[0],
-			slackAttachment.Fields[1],
-			{
-				Title: "Approvers",
-				Value: fmt.Sprintf("%s %s", constants.PipelineRequestUpdateEmoji[requestType], slackAttachment.Fields[2].Value),
-			},
-		}
-
-		model.ParseSlackAttachment(post, []*model.SlackAttachment{slackAttachment})
-		if _, err := p.API.UpdatePost(post); err != nil {
-			if _, DMErr := p.DM(mattermostUserID, constants.GenericErrorMessage, true); DMErr != nil {
-				p.API.LogError("Failed to DM", "Error", DMErr.Error())
-			}
-
-			p.API.LogError("Error in updating post", "Error", err.Error())
-			p.handleError(w, r, &serializers.Error{Code: http.StatusInternalServerError, Message: err.Error()})
+		if updatePipelineReleaseApprovalPostErr := p.UpdatePipelineReleaseApprovalPost(requestType, postActionIntegrationRequest.PostId, mattermostUserID); updatePipelineReleaseApprovalPostErr != nil {
+			p.handlePipelineApprovalRequestUpdateError(constants.GenericErrorMessage, mattermostUserID, updatePipelineReleaseApprovalPostErr)
+			p.handleError(w, r, &serializers.Error{Code: http.StatusInternalServerError, Message: updatePipelineReleaseApprovalPostErr.Error()})
 			return
 		}
-
 	case http.StatusBadRequest:
-		if _, DMErr := p.DM(mattermostUserID, fmt.Sprintf(constants.ErrorUpdatingNonPendingPipelineRequest, approvalID), true); DMErr != nil {
-			p.API.LogError("Failed to DM", "Error", DMErr.Error())
+		pipelineApprovalDetails, statusCode, getApprovalDetailsErr := p.Client.GetApprovalDetails(organization, projectName, mattermostUserID, approvalID)
```

Comment:
```suggestion
		pipelineApprovalDetails, statusCode, err := p.Client.GetApprovalDetails(organization, projectName, mattermostUserID, approvalID)
```
IMO, using such big names for errors here is reducing the simplicity and readability of the code here.

Reply:
Yes it is for readability and also to prevent chances of shadow declaration with variable name `err`

Reply:
I think you used such big names for increasing readability but IMO, it is decreasing the readability and simplicity of the code here. We can always avoid shadow declaration of `err` by using linter but we should keep the code as simple as possible.